### PR TITLE
feat(extension): implement fuzzy search in fetch stake pool slice and…

### PR DESCRIPTION
**Pull request by vetalcore**
… utils

# Checklist

- [x] JIRA - [LW-10378](https://input-output.atlassian.net/browse/LW-10378)
- [x] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

[LW-10378]: https://input-output.atlassian.net/browse/LW-10378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser-extension-wallet/src/stores/slices/stake-pool-search-slice.ts [hunk 1]
```diff
@@ -18,21 +18,21 @@ const fetchStakePools =
...
     set({ stakePoolSearchResultsStatus: StateStatus.LOADING });
 
-    let filtersValues = [];
+    let filtersValues: Wallet.QueryStakePoolsArgs['filters'] = {};
     try {
       const poolId: Wallet.Cardano.PoolId = Wallet.Cardano.PoolId(searchString);
-      filtersValues = [{ id: poolId }];
+      filtersValues = {
+        identifier: {
+          _condition: 'or',
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/stores/slices/stake-pool-search-slice.ts:26]: there shouldn't be a need for this `or` as it's the only possible field, you can probably safely remove this

File: apps/browser-extension-wallet/src/views/browser-view/features/staking/utils.ts [hunk 2]
```diff
@@ -7,14 +7,21 @@ export const fetchPoolsInfo = async ({
   searchString: string;
   stakePoolProvider: Wallet.StakePoolProvider;
 }): Promise<Wallet.StakePoolSearchResults['pageResults']> => {
+  let filtersValues: Wallet.QueryStakePoolsArgs['filters'] = {};
+  try {
+    const poolId: Wallet.Cardano.PoolId = Wallet.Cardano.PoolId(searchString);
+    filtersValues = {
+      identifier: {
+        _condition: 'or',
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/features/staking/utils.ts:15]: same comment about removing this one

**Review by mchappell — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/stores/slices/stake-pool-search-slice.ts (hunk 1 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/stores/slices/stake-pool-search-slice.ts]: done here a917106da9d9c0643cf82fbc7d45e226e2960b6d

File: apps/browser-extension-wallet/src/views/browser-view/features/staking/utils.ts (hunk 2 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/staking/utils.ts]: done here https://github.com/input-output-hk/lace/commit/a917106da9d9c0643cf82fbc7d45e226e2960b6d

**Review by mchappell — APPROVED** [review]
Great change 💪

**Review by mchappell — APPROVED** [review]

**Review by wklos-iohk — APPROVED** [review]

**Pull request merged**